src/blockMonitor/blockCache.ts

```diff
@@ -111,90 +107,104 @@ export class BlockCache<TBlock extends IBlockStub> implements ReadOnlyBlockCache
         return this.isEmpty || this.hasBlock(block.parentHash) || block.number === this.minHeight;
     }
 
-    // Processes all the detached blocks at the given height, moving them to blocksByHeight if they are now attached.
-    // If so, add them and repeat with the next height (as some blocks might now have become attached)
-    private processDetached(height: number) {
-        const blockHashesAtHeight = this.blockHashesByHeight.get(height) || new Set();
-        const blockHashesToAdd = [...blockHashesAtHeight].filter(h => this.detachedBlocksByHash.has(h));
+    private async emitNewBlockEvent(block: TBlock) {
+        for (const listener of this.newBlockListeners) {
+            await listener(block);
+        }
+    }
 
-        for (const blockHash of blockHashesToAdd) {
-            const block = this.detachedBlocksByHash.get(blockHash)!;
+    // Processes all the detached blocks at the given height, marking them as attached if necessary.
+    // If there are any, add them and repeat with the next height (as some blocks might now have become attached)
```

Comment:
I assumed that the /** */ comments was for documentation of everything public and it could be picked up by some automated documentation generator to produce a developer manual. No problem switching to /** */ by default for all methods from now on, though.